ZenPacks/zenoss/Microsoft/Windows/datasources/PerfmonDataSource.py

```diff
@@ -231,12 +231,14 @@ def start(self, command_lines):
         create an appropriate set of commands.
         """
         deferreds = []
-        if self.num_commands != len(command_lines):
-            self.commands = self._create_commands(len(command_lines))
+        if self.num_commands != len(self.commands):
+            self.num_commands = len(self.commands)
```

Comment:
I can't quite follow this- so the idea here is to re-create self.commands, but only if self.num_commands has changed- it looks like the only way that happens is if remove_corrupt_counters does it?  Is that right so far?

I think i'd like to see a comment here explaining that, if so..

And it looks like self._create_commands sets self.num_commands to the value that was passed in (in this case, self.num_commands's existing value), which is.. confusing.   Because that doesn't really make sense, if it is based on the length of self.commands already..  So this seems like it would all do nothing?    

I think i'm missing something- i don't reall yknow what command_lines is either :)

Reply:
well, the only part i changed here was resetting self.num_commands to the new number of commands.  the self._create_commands has always been there.

for the windows perfmon get-counter command, there is a line length of 8192.  if a user has added a new template or created their own with more counters, this could cause the line length to go past that limit and we'd need to create  a new command.  or in the case of the corrupt counters it could lower it so we don't need as many commands.